embassy-nrf/src/uarte.rs

```diff
@@ -173,6 +173,61 @@ impl<'d, T: Instance> Uarte<'d, T> {
         (self.tx, self.rx)
     }
 
+    /// Split the Uarte into a transmitter and receiver that will
```

Comment:
The following implementation was mostly taken from what was the constructor for `UarteWithIdle`.